sopel/modules/clock.py

```diff
@@ -54,14 +54,29 @@ def setup(bot):
 
 @module.commands('t', 'time')
 @module.example('.t America/New_York')
+@module.example('.t Exirel')
+@module.example('.t #sopel')
 def f_time(bot, trigger):
-    """Returns the current time."""
+    """Returns the current time.
+
+    The command takes an optional parameter: it can be either a timezone, a
+    nick, or a channel.
+    """
     if trigger.group(2):
-        zone_name = trigger.group(2).strip()
+        argument = trigger.group(2).strip()
+        channel_or_nick = tools.Identifier(argument)
+
         try:
-            zone = validate_timezone(zone_name)
+            if channel_or_nick in bot.users:
+                zone = get_timezone(bot.db, bot.config, None, channel_or_nick,
+                                    trigger.sender)
+            elif channel_or_nick in bot.channels:
+                zone = get_timezone(bot.db, bot.config, None, None,
+                                    channel_or_nick)
```

Comment:
Same concern here. Falling back on the bot's default timezone is potentially misleading if the channel doesn't have one set for itself. Again, I think an error is better.